metrics/metrics-core/src/main/java/io/servicecomb/metrics/core/health/DefaultMicroserviceHealthChecker.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicecomb.metrics.core.health;
+
+import javax.ws.rs.core.Response.Status;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.servicecomb.foundation.common.utils.JsonUtils;
+import io.servicecomb.metrics.common.DefaultHealthCheckExtraData;
+import io.servicecomb.metrics.common.HealthCheckResult;
+import io.servicecomb.metrics.common.HealthChecker;
+import io.servicecomb.serviceregistry.RegistryUtils;
+import io.servicecomb.serviceregistry.api.registry.Microservice;
+import io.servicecomb.serviceregistry.api.registry.MicroserviceInstance;
+import io.servicecomb.swagger.invocation.exception.InvocationException;
+
+public class DefaultMicroserviceHealthChecker implements HealthChecker {
+
+  private static Logger logger = LoggerFactory.getLogger(DefaultMicroserviceHealthChecker.class);
+
+  @Override
+  public String getName() {
+    return "default";
+  }
+
+  @Override
+  public HealthCheckResult check() {
+    return new HealthCheckResult(true, "", getExtraData());
+  }
+
+  private String getExtraData() {
+    try {
+      Microservice microservice = RegistryUtils.getMicroservice();
+      MicroserviceInstance instance = RegistryUtils.getMicroserviceInstance();
+      return JsonUtils.writeValueAsString(new DefaultHealthCheckExtraData(
```

Comment:
is it safe to report this to remote?

and what's the relationship between this health status and which in serviceCenter?

in very early of CSE, we had a plan for this.
provider health check interface, then business report their status, if not health, then SDK will stop heartBeat to SC or report status to SC.

Reply:
I think report this is OK ,or we can add a switch with default false?
Our config center may not use ServiceCenter,Apollo is coming. And Health check is more wide mechanism also can used for check DB,Microservice or other related resource,one Microservice can have many HealthChecker;
So I think "if not health, then SDK will stop heartBeat to SC or report status to SC" is a Ops policy and we can support later

Reply:
First Configure center != Service Center. 
ServiceCenter is more like a service registry which has the .  
The Health check can work with other system admin tool to help the admin check find out the system failure. As it is working in the pull mode by default, we don't report the service status to the Service Center now.

Reply:
oh,no ,sorry I made a mistake，comment fixed